client/src/Register.js

```diff
@@ -22,14 +25,26 @@ function Register() {
     const newUser = {...user};
     newUser[event.target.name] = event.target.value;
     setUser(newUser);
+    setUserType(userType);
```

Comment:
So, this isn't going to work as expected... the new value of userType from the select change will be in event.target.value. I'd remove this new line here and make changes elsewhere...